### PR TITLE
GEOMESA-1701 Spark RPC endpoint for kryo schema

### PR DESCRIPTION
**Pull request by tkunicki**
Signed-off-by: Tom Kunicki <tom.kunicki@ccri.com>

### CODE REVIEW

**Comment by vpipkt** [comment]
Tested branch using Spark 2.0.2 on Yarn; I set only the `spark.serializer` and `spark.kryo.registrator` config keys. Created an accumulo SpatialRDDProvider, got an RDD[SImpleFeatures] and `take(n)` works fine (prints FID); `take(n).foreach(println(_.getAttributes))` works as expected. 

In mapPartitions, I created a new simple feature type and mapped existing SFs into it, repartitioned up and down; and collected to driver. We can see the different SFTs from the different executors. A call to GeoMesaSparkKryoRegistrator.getTypes shows the new SFTs were added. Logging shows the putType and getType requests are very fast, 2 or 3 milliseconds.

User / analyst perspective before this PR: not sure why this doesn't work or why i am using some init incantation 
Now: simple features and simple feature types just work.

**Comment by jahhulbert-ccri** [comment]
@vpipkt did you look through the code?

**Comment by vpipkt** [comment]
I skimmed through but not very closely.

File: geomesa-spark/geomesa-spark-core/src/main/scala/org/locationtech/geomesa/spark/GeoMesaSparkKryoRegistratorEndpoint.scala [hunk 1]
```diff
@@ -0,0 +1,123 @@
+/***********************************************************************
+* Copyright (c) 2013-2016 Commonwealth Computer Research, Inc.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Apache License, Version 2.0
+* which accompanies this distribution and is available at
+* http://www.opensource.org/licenses/apache2.0.php.
+*************************************************************************/
+
+package org.apache.spark.geomesa
```
Comment (jnh5y) [inline geomesa-spark/geomesa-spark-core/src/main/scala/org/locationtech/geomesa/spark/GeoMesaSparkKryoRegistratorEndpoint.scala]: The path and the package don't match.  I'd move this to the expected directory structure.

File: geomesa-spark/geomesa-spark-core/src/main/scala/org/locationtech/geomesa/spark/GeoMesaSparkKryoRegistrator.scala [hunk 2]
```diff
@@ -61,33 +62,39 @@ object GeoMesaSparkKryoRegistrator {
 
   private val typeCache = new ConcurrentHashMap[Int, SimpleFeatureType]()
 
+  GeoMesaSparkKryoRegistratorEndpoint.init()
```
Comment (jnh5y) [inline geomesa-spark/geomesa-spark-core/src/main/scala/org/locationtech/geomesa/spark/GeoMesaSparkKryoRegistrator.scala]: Is this idempotent?

File: geomesa-spark/geomesa-spark-core/src/main/scala/org/locationtech/geomesa/spark/GeoMesaSparkKryoRegistratorEndpoint.scala [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+  def init(): Unit = {
+    Option(SparkEnv.get).foreach {
+      sparkEnv =>
+        sparkEnv.executorId match {
+          case SparkContext.DRIVER_IDENTIFIER =>
+            val rpcEnv = sparkEnv.rpcEnv
+            Try(rpcEnv.setupEndpoint(EndpointName, new KryoEndpoint(rpcEnv))) match {
+              case Success(ref) => logger.info(s"$EndpointName rpc endpoint registered on driver ${ref.address}")
+              // Can't test if endpoint already exists using available abstractions, failure expected if already bound
+              case Failure(e) => logger.debug(s"$EndpointName rpc endpoint registration failed", e)
```
Comment (jnh5y) [inline geomesa-spark/geomesa-spark-core/src/main/scala/org/locationtech/geomesa/spark/GeoMesaSparkKryoRegistratorEndpoint.scala]: Maybe add 'may already be registered"...

**Pull request merged**